### ISSUE TITLE
question - keep receiving errors

### ISSUE THREAD

**Issue by tyler-g** [erossi/m5e_rfid#1]
Hi @erossi , first thank you for this great library.   Unfortunately i can't seem to get it to work.

I'm using a CHIP device (https://getchip.com/pages/chip).  I've got an M5e-CZ RFID reader,  connected to the CHIP  with power in (3.3V),  GND,  and UART RX + TX.

For me the device is `/dev/ttyS0`   since on the CHIP that is the device for UART serial communication.

I'm able to "connect" to the serial device, but doesn't seem to have any communication.  Everything is giving me an error, and since the code is mostly in try/except blocks, I'm not able to print out any debugging info.   Forgive me, i'm pretty new to python.

**Comment by erossi** [comment]
Hi Tyler,
Did you check, as by chip docs, that the serial port is OK and no daemons are interfering with it? http://docs.getchip.com/chip.html#uart on the bottom. Do you have an error also on the boot (04) and get_version (03) command? Can you please copy-paste the error?
Thanks for reporting.
Bye
Enrico

**Comment by tyler-g** [comment]
Hi Enrico,
 I am not sure about daemons interfering with it,  but I don't think they are because I did a little test.  I tried jumping the RX pin on the CHIP to the TX pin.   Then as root I echo to `/dev/ttyS0`  and I can see the response if I listen to `/dev/ttyS0`.  So as far as I know,  `/dev/ttyS0` is the device for the UART RX/TX pins on the right header of the chip (pins 43 and 45).

I had to modify your script a little bit because maybe it wasn't acting right with my version of python.  I'm new to python though, so was probably my own fault.  

Anyway, I did get your script to run,  and if I send an '03' command (which is translating to `'\xFF\x00\x03\x1D\x0C'`, I just get an empty response after the timeout expires.  Same thing with '04' or any of the commands I tried.

**Comment by erossi** [comment]
Tyler,
The software is written for python3 (3.5). I think that on CHIP you have to install it and it should be 3.4.
Can you tell me which version are you using?

**Comment by tyler-g** [comment]
You're right it was 3.4.  I'm installing 3.5 now.   In the meantime I had modified a few of the commands to work on 3.4. 

  I'm happy to say though that I'm now getting responses from my RFID reader.  It looks like the problem was that the connector I was using wasn't staying secure.  I ended up soldering the connections and now I'm seeing responses,  but not like the ones in your example.  I'm looking through the manual to try to understand how to interpret them.

**Comment by tyler-g** [comment]
It sees as though I'm able to send a command, and I get a response,  but any command I send after that gets an empty response.

If I send `04`  I am getting back `040101c545 `.  If I then send `03` I get empty response. 

However if I close the script and run it again,  with `03`  I get:
`00 07 10 2300 01000003 20080620 01003769 00000010 1be1`

I believe it might have something to do with the flushing, but thats just a guess, hmm

**Comment by tyler-g** [comment]
update:  it turns out the CHIP's getty service needs to be disabled, otherwise it interferes with /dev/ttyS0 and was causing bad data.  For others reference,  it can be disabled by:

`sudo systemctl stop serial-getty@ttyS0.service`

You can also disable the service so you don't need to run the above command after every reboot.


Now, I'm able to run set and get commands .. everything matches what your examples show, except for actually reading a tag.   Any time I try to read a tag (single or multiple),  I get a returned status of 
`7F01` . 

According to the manual (http://www.dpie.com/manuals/rfid/M5eFamilyDevGuide_Jan12.pdf), this means `An unexpected Internal Error has occurred`  :(   and says to email tech support.  When I email them they tell me its out of warranty and can't help me.

I even tried downloading from source, and making their ArbSer tool (command line interface for updating firmware and sending commands, etc.).  I was able to update to the latest firmware, and yet still get this `7F01` error when trying to read tags.

**Comment by tyler-g** [comment]
fyi I got it working,  and works great.  It turns out to be a power supply problem that was causing insufficient current.

Theres a few update suggestions I had for your script.  Mostly to do with timeouts  and making it run a bit faster.  Will be happy to do a PR later when I get the time.   Thanks for the help

**Comment by erossi** [comment]
Great!
Please feel free to improve it. It was a quick script to test the device.
Thanks
E.
